src/fides/api/alembic/migrations/versions/4ebe0766021b_add_connection_type_to_dataset_namespace.py

```diff
@@ -0,0 +1,51 @@
+"""Add connection_type to dataset namespace meta
+
+Revision ID: 4ebe0766021b
+Revises: 36ad82edb38e
+Create Date: 2024-10-16 15:25:08.544946
+
+"""
+
+import sqlalchemy as sa
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "4ebe0766021b"
+down_revision = "36ad82edb38e"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # Populate the connection_type field in the namespace meta for all datasets
+    # that have a namespace key in their fides_meta json field.
+
+    # If the namespace dict has a project_id key, we know it's a BigQuery dataset.
+    op.execute(
+        """
+        update ctl_datasets
+        set fides_meta = jsonb_set(fides_meta::jsonb, '{namespace, connection_type}', '"bigquery"', true)
+        where (fides_meta::jsonb ? 'namespace') and ((fides_meta::jsonb ->> 'namespace')::jsonb ? 'project_id');
+        """
+    )
+
+    # If the namespace dict has a database_instance_id, we know it's an RDS MySQL dataset.
+    op.execute(
```

Comment:
I'm missing context from the ticket, sorry, but do we care about any other dialects? We'll fill those in later?

Reply:
My fidesplus PR https://github.com/ethyca/fidesplus/pull/1673 (not ready for review yet) makes the necessary changes to start storing it for all datasets created through D&D . This migration just takes advantage of the fact that BQ and RDS MySQL were both already using the `namespace` key so we have a reliable way to identify existing datasets from these integrations. for other integrations we'll start storing the connection_type from now on but we don't have a way to populate their values for existing datasets